codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperFuelWorker.res

```diff
@@ -25,8 +27,20 @@ let makeWorkerConfigOrThrow = (~contracts: array<Types.fuelContractConfig>, ~cha
   let contractByEvent = Utils.WeakMap.make()
   let nonWildcardLogDataRbsByContract = Js.Dict.empty()
   let wildcardLogDataRbs = []
-  let contractsWithMint = Utils.Set.make()
-  let hasWildcardMint = ref(false)
+
+  // This is for non-LogData events, since they don't have rb filter and can be grouped
+  let nonLogDataReceiptTypesByContract = Js.Dict.empty()
+  let nonLogDataWildcardReceiptTypes = []
+
+  let addNonLogDataWildcardReceiptTypes = (receiptType: Fuel.receiptType) => {
+    nonLogDataWildcardReceiptTypes->Array.push(receiptType)->ignore
+  }
+  let addNonLogDataReceiptType = (contractName, receiptType: Fuel.receiptType) => {
+    switch nonLogDataReceiptTypesByContract->Utils.Dict.dangerouslyGetNonOption(contractName) {
+    | None => nonLogDataReceiptTypesByContract->Js.Dict.set(contractName, [receiptType])
+    | Some(receiptTypes) => receiptTypes->Array.push(receiptType)->ignore // Duplication prevented by EventRouter
```

Comment:
Can we not be cautious and use a set or do a runtime check for duplicate since this is only run on startup?

I understand it's already been checked but if order of function calls moves later on etc it's nice to have protection against these kind of bugs.

Reply:
We have tests for this.